### PR TITLE
adds a debugging flag for bundle parser

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

### Description
adds a `_debug_` keyword, so that when it's set to true, the program stops at the component init stage and enter the debug mode , usage:

```yaml
inferer:
  _target_: SlidingWindowInferer
  _debug_: true
  roi_size: [96, 96, 96]
  sw_batch_size: 4
  overlap: 0.5
```

alternatively at the CLI set `MONAI_DEBUG_CONFIG=1`, for example:

```bash
MONAI_DEBUG_CONFIG=1 python -m my_test_5 run --config_file=my_test_5.yaml
```

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by wyli** [comment]
two ways to enter the debug mode:

- adds `_debug_: true` to the component in the config
- run with a system `MONAI_DEBUG_CONFIG` flag when starting a bundle.configparser from CLI:  
```bash
MONAI_DEBUG_CONFIG=0 python -m my_test_5 run --config_file=my_test_5.yaml
```

File: monai/bundle/config_item.py [hunk 1]
```diff
@@ -365,7 +364,15 @@ def evaluate(self, globals: Optional[Dict] = None, locals: Optional[Dict] = None
...
                     warnings.warn(f"the new global variable `{k}` conflicts with `self.globals`, override it.")
                 globals_[k] = v
-        return eval(value[len(self.prefix) :], globals_, locals)
+        if not run_debug:
+            return eval(value[len(self.prefix) :], globals_, locals)
+        warnings.warn(
+            f"\n\npdb: value={value}\n"
+            f"See also Debugger commands documentation: https://docs.python.org/3/library/pdb.html\n"
+        )
+        import pdb
```
Comment (Nic-Ma) [inline monai/bundle/config_item.py:373]: Should we import `pdb` at the beginning of the file?

Thanks.

File: monai/utils/module.py [hunk 2]
```diff
@@ -220,6 +228,14 @@ def instantiate(path: str, **kwargs):
...
         raise ModuleNotFoundError(f"Cannot locate class or function path: '{path}'.")
     try:
+        if kwargs.pop("_debug_", False) or run_debug:
+            warnings.warn(
+                f"\n\npdb: instantiating component={component}\n"
+                f"See also Debugger commands documentation: https://docs.python.org/3/library/pdb.html\n"
+            )
+            import pdb
+
+            pdb.set_trace()
```
Comment (Nic-Ma) [inline monai/utils/module.py:238]: Is it possible to add some unit tests to cover the behavior here?
I didn't test `pdb` locally, I think it's an interactive operation?

Thanks.

File: monai/bundle/config_item.py (hunk 1 above)
Comment (wyli) [reply monai/bundle/config_item.py:373]: don't want to introduce this tool to the module, it's mostly not used, if we find it's frequently I'll update this.

**Review by Nic-Ma — APPROVED** [review]
Thanks for the quick update, looks good to me.
Could you please help also update the `get_started` tutorial of bundle to show this feature?

Thanks in advance.

**Pull request merged**